pkg/k3d/cluster.go

```diff
@@ -82,9 +86,59 @@ func CreateCluster(ctx context.Context, manifest *manifest.InstallationManifest,
 		log.Println(err)
 	}
 
+	if params.CpuLimit != "" {
+		cpuLimit := getCPUsLimit(params.CpuLimit)
+		if err := applyCpuLimit(strconv.Itoa(cpuLimit)); err != nil {
+			log.Fatalf("Failed to apply CPU limit: %v", err)
+		}
+
+		log.Infof("CPU limit applied to all k3d containers: %d\n", cpuLimit)
+	}
+
 	return
 }
 
+func applyCpuLimit(cpuLimit string) error {
+	containers, err := getK3dContainers()
+	if err != nil {
+		return fmt.Errorf("failed to get k3d containers: %w", err)
+	}
+
+	log.Infof("K3DContainers to update cpu limit: %v\n", containers)
+	for _, container := range containers {
+		if err := updateContainerCPU(container, cpuLimit); err != nil {
+			log.Printf("Failed to update CPU limit for container %s: %v", container, err)
+		} else {
+			fmt.Printf("CPU limit applied to container %s\n", container)
+		}
+	}
+
+	return nil
+}
+
+func getK3dContainers() ([]string, error) {
+	cmd := exec.Command("docker", "ps", "--filter", "name=k3d", "--format", "{{.ID}}")
```

Comment:
consider the case user has two k3d cluster

Reply:
Changed to `k3d-tensorleap`